backend/routers/routes/posts.py

```diff
@@ -25,7 +28,59 @@ async def get_posts(c_id: str):
     return {"posts": posts}
 
 
-# @post_router.get("/{post_id}", response_model=GetPostResponse)
-# async def get_post(post_id: str):
-#     post = post_controller.get_post(post_id)
+"""
+Missing Endpoints:
+- Get singlar post 
+    TODO
+    - Get edit history
+    - Get event history
+- Edit post DONE
+- Delete post DONE
+- View post DONE
+- Like post DONE
+- Get post metadata TODO
+
+- Missing DB
+    - post_impressions materialized view TODO
+"""
+
+
+@post_router.get("/{p_id}", response_model=GetPostResponse)
+async def get_post(c_id: str, p_id: str):
+    post = post_controller.get_post(c_id, p_id)
+    return {"post": post}
+
+
+@post_router.patch("", response_model=GeneralResponse)
+async def update_post(c_id: str, request: Request, post_edit_info: CreatePostEditRequest):
+    user_id = request.state.user_id
+    post_controller.update_post(c_id, user_id, post_edit_info)
+
+    return {"msg": "Post edited successfully"}
+
+
+@post_router.delete("/{post_id}", response_model=GeneralResponse)
+async def delete_post(c_id: str, post_id: str, request: Request):
+    user_id = request.state.user_id
+    post_controller.delete_post(c_id, user_id, post_id)
+
+    return {"msg" : "Post edited successfully"}
+
+
+@post_router.put("/{post_id}/view", response_model=GeneralResponse)
+async def view_post(c_id: str, post_id: str, request: Request):
+    user_id = request.state.user_id
+    post_controller.view_post(c_id, user_id, post_id)
+    return {"msg" : "Post viewed"}
+
+
+@post_router.post("/{post_id}/like", response_model=GeneralResponse)
+async def like_post(c_id: str, post_id: str, request: Request):
+    user_id = request.state.user_id
+    post_controller.like_post(c_id, user_id, post_id)
+    return {"msg" : "Post liked"}
```

Comment:
Can we change this to a more high-level way:
`..../{post_id}/events/like`

small nitpick